### PR TITLE
Add doc to our Wire extensions/output

### PR DESCRIPTION
**Pull request by oldergod**
So that we show something when DSLing within gradle.

image

### CODE REVIEW

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt [hunk 1]
```diff
@@ -207,24 +207,36 @@ open class WireExtension(
     return result
   }
 
+  /**
+   * Defines a Java target. See [com.squareup.wire.schema.JavaTarget]
```
Comment (Egorand) [inline wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt:211]: Comments should be sentences!

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt [hunk 2]
```diff
@@ -277,16 +295,19 @@ open class WireExtension(
     internal val includes = mutableListOf<String>()
     internal val excludes = mutableListOf<String>()
 
+    /** Sets a directory. Example: "src/main/proto". */
     fun srcDir(dir: String) {
       isEmpty = false
       sourceDirectoriesAndLocalJars += project.file(dir)
     }
 
+    /** Sets one or more directory. */
```
Comment (Egorand) [inline wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt:304]: Shouldn't it be plural?

```suggestion
    /** Sets one or more directories. */
```

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt [hunk 3]
```diff
@@ -323,10 +348,18 @@ open class WireExtension(
       project.dependencies.add("protoProjectDependenciesJvm", dependencyNotation)
     }
 
+    /**
+     * If set, only the files being defined as included will be treated.
```
Comment (Egorand) [inline wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt:352]: nit:

```suggestion
     * If set, only the files defined as included will be processed.
```

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt [hunk 4]
```diff
@@ -323,10 +348,18 @@ open class WireExtension(
...
+     * Example: "com/example/important.proto".
+     */
     fun include(vararg includePaths: String) {
       includes += includePaths
     }
 
+    /**
+     * If set, all the files being defined as excluded will be ignored.
```
Comment (Egorand) [inline wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt:360]: nit:

```suggestion
     * If set, all the files defined as excluded will be ignored.
```

**Review by Egorand — APPROVED** [review]

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt (hunk 1 above)
Comment (oldergod) [reply wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireExtension.kt]: déjà vu !!!

Thank you, all addressed.

**Pull request merged**